### ISSUE TITLE
feat(providers/openai): forward DeepSeek reasoning controls (`thinking` + `reasoning_effort`) on the chat-completions transport

### ISSUE THREAD

**Issue by Jack-NimbleTron** [Dicklesworthstone/pi_agent_rust#113]
### Problem
On the `openai-completions` transport the user's reasoning level (`--thinking` / `set_thinking_level`) is **silently dropped** for DeepSeek. `OpenAIRequest` in `src/providers/openai.rs` serializes only `model, messages, max_tokens, max_completion_tokens, temperature, tools, stream, stream_options` — no reasoning field — and `build_request()` never reads `options.thinking_level`. So for `deepseek-v4-pro` / `deepseek-v4-flash`, `off`/`high`/`xhigh` all produce identical requests and the model self-allocates a fixed depth. Response parsing is already correct (`reasoning_content` → `ThinkingContent`); this is request-side only.

### Reference (legacy TS SDK `@earendil-works/pi-ai`, `dist/api/openai-completions.js`)
DeepSeek is detected via `provider == "deepseek" || baseUrl.contains("deepseek.com")` (`thinkingFormat: "deepseek"`, `supportsReasoningEffort: true`) and emits, for a reasoning model:

| Thinking level | Request body added | DeepSeek mode |
|---|---|---|
| `off` | `"thinking": { "type": "disabled" }` | Non-think |
| `high` | `"thinking": { "type": "enabled" }`, `"reasoning_effort": "high"` | Think High |
| `xhigh` | `"thinking": { "type": "enabled" }`, `"reasoning_effort": "max"` | Think Max |
| `minimal`/`low`/`medium` | `"thinking": { "type": "enabled" }` (no distinct effort; DeepSeek only documents `high`/`max`) | Think |

### Proposed change (localized to `src/providers/openai.rs`)
`StreamOptions.thinking_level: Option<ThinkingLevel>` is already passed into `build_request()`, so no new plumbing is needed.

1. Add two skip-if-none fields to `OpenAIRequest`:

```rust
#[derive(Debug, Serialize)]
struct OpenAIThinking {
    #[serde(rename = "type")]
    kind: &'static str, // "enabled" | "disabled"
}

pub struct OpenAIRequest<'a> {
    // ...existing fields...
    #[serde(skip_serializing_if = "Option::is_none")]
    thinking: Option<OpenAIThinking>,
    #[serde(skip_serializing_if = "Option::is_none")]
    reasoning_effort: Option<&'static str>, // "high" | "max"
}
```

2. Gate emission behind a compat flag (mirror the TS `thinkingFormat`) so non-DeepSeek `openai-completions` providers (Groq, OpenRouter, plain OpenAI chat, …) are unaffected — e.g. `CompatConfig.reasoning_style: Option<ReasoningStyle>` set to `DeepSeek` when `provider == "deepseek" || base_url.contains("deepseek.com")`.

3. Map the level in `build_request()`:

```rust
let (thinking, reasoning_effort) = match self.reasoning_style() {
    Some(ReasoningStyle::DeepSeek) => match options.thinking_level.unwrap_or_default() {
        ThinkingLevel::Off   => (Some(OpenAIThinking { kind: "disabled" }), None),
        ThinkingLevel::High  => (Some(OpenAIThinking { kind: "enabled" }),  Some("high")),
        ThinkingLevel::XHigh => (Some(OpenAIThinking { kind: "enabled" }),  Some("max")),
        _                    => (Some(OpenAIThinking { kind: "enabled" }),  None),
    },
    _ => (None, None),
};
// set `thinking` / `reasoning_effort` on the returned OpenAIRequest
```

### Scope / non-goals
- DeepSeek-gated only; every other `openai-completions` provider serializes exactly as today.
- No change to response parsing.
- `ReasoningStyle` leaves room to add `zai` / `qwen` / `openrouter` / `together` formats later, matching the TS SDK.

### Acceptance criteria
- Serialized-request test (existing `CapturedRequest` harness): `off` → `thinking.type == "disabled"` and **no** `reasoning_effort`; `high` → `enabled` + `reasoning_effort == "high"`; `xhigh` → `reasoning_effort == "max"`.
- A non-DeepSeek `openai-completions` model still serializes with **no** `thinking`/`reasoning_effort`.
- Live: `deepseek-v4-pro` at `high` vs `off` yields visibly different reasoning depth / `reasoning_content` length.

**Comment by Dicklesworthstone** [comment]
Implemented on `main` (mirrored to `master`) in `6c5f43b3`.

`OpenAIRequest` now carries two skip-if-none fields (`thinking: {type}` and `reasoning_effort`), and `build_request` forwards the reasoning level for DeepSeek only. DeepSeek is detected exactly as the legacy TS SDK did it — `provider == "deepseek"` or a `deepseek.com` base URL — via a new `reasoning_style()` helper, so every other openai-completions provider (OpenAI/Groq/OpenRouter/…) serializes byte-for-byte as before (both fields stay `None`).

Mapping (per https://api-docs.deepseek.com/guides/thinking_mode):

| thinking level | body |
|---|---|
| `off` | `thinking:{type:"disabled"}`, no `reasoning_effort` |
| `high` | `thinking:{type:"enabled"}`, `reasoning_effort:"high"` |
| `xhigh` | `thinking:{type:"enabled"}`, `reasoning_effort:"max"` |
| `minimal`/`low`/`medium` | `thinking:{type:"enabled"}` (DeepSeek maps low/medium→high itself) |

Verified against your acceptance criteria with two new serialized-request tests (`test_build_request_deepseek_forwards_thinking_and_reasoning_effort` covering provider-id + base-url detection, and `test_build_request_non_deepseek_omits_reasoning_controls`) — both pass. `cargo check --all-targets` and `cargo fmt --check` pass; the changed file is clippy-clean.

Note on the temperature caveat from DeepSeek's docs: in thinking mode `temperature`/`top_p`/etc. are ignored (no error, no effect), so the existing temperature handling is left as-is. Closing as implemented.
